OSDOCS-4883: Document OSD/ROSA managed Ingress improvements (#60887)
openshift/openshift-docs: applications/deployments/osd-config-custom-domains-applications.adoc

toc::[]

[NOTE]
====

> Starting with {product-title} 4.14, the Custom Domain Operator is deprecated, and you can manage Ingress with the Ingress Operator. The functionality is unchanged for {product-title} 4.13 and earlier versions. 

Reviewer: ```suggestion
Starting with {product-title} 4.14, the Custom Domain Operator is deprecated. To manage Ingress in {product-title} 4.14, use the Ingress Operator. The functionality is unchanged for {product-title} 4.13 and earlier versions. 
```
